src/realm/sync/noinst/migration_store.cpp

```diff
@@ -101,9 +103,21 @@ bool MigrationStore::is_migrated()
     return m_state == MigrationState::Migrated;
 }
 
-std::string_view MigrationStore::get_query_string()
+std::optional<std::string_view> MigrationStore::get_migrated_partition()
 {
     std::lock_guard lock{m_mutex};
+    if (m_state != MigrationState::Migrated) {
+        return std::nullopt;
+    }
+    return m_migrated_partition;
```

Comment:
I'm just realizing that this function and get_query_string() are returning a string_view which is an unowned view of a std::string protected by a lock but don't guarantee that the lock is held after returned, so it's possible to call get_migrated_partition() and have the returned string_view pointing to free'd or otherwise invalid memory after return.

I think both this and get_query_string() need to return a std::optional<std::string>